Automattic/wp-calypso
# Remove isSiteUsingCoreSiteEditor selector from sidebar and theme options

#### Changes proposed in this Pull Request

**Theme options**

- Replace `isSiteUsingCoreSiteEditor` selector with `withBlockEditorSettings`
- Update customize theme action so it directs to the Site editor when FSE is enabled

<img width="743" alt="Screen Shot 2021-11-03 at 17 23 02" src="https://user-images.githubusercontent.com/1699996/140225970-2166efaa-48e9-4da6-967b-a3afbeab45d3.png">

**Legacy sidebar**

- Replace `isSiteUsingCoreSiteEditor` selector with `withBlockEditorSettings`
- Update logic for Universal themes, so both Site editor and Customizer are accessible

**Note:** I don't think the legacy sidebar is in use any longer, except for non-Atomic Jetpack sites (which this PR does not change), but I'm updating it in case that's wrong.

<img width="274" alt="Screen Shot 2021-11-02 at 16 37 04" src="https://user-images.githubusercontent.com/1699996/140226053-0121ef6d-793d-496d-b1ab-d599aa4e1068.png">


#### Testing instructions

Set up a site so that FSE is active (enroll in the FSE beta through horizon.wordpress.com/new and make sure a universal theme, like Quadrat is active)

**Theme options**

- Go to `/themes/{site}`
- See that primary action for the current theme it now "Edit" and links to the Site editor when FSE is active
- The action remains "Customize" and links to the Customizer or page editor for the Homepage for non-FSE sites

**Legacy Sidebar**

- Show the legacy sidebar by adding `?disable-nav-unification` to the url when loading Calypso
- See that the Customizer and the Site editor are available for FSE enabled themes

Partially addresses #57625
